### PR TITLE
Remove DefaultURLResolverService and its implementation

### PR DESCRIPTION
**Pull request by ChamathNS**
### Proposed changes in this pull request

$subject.

With #2851 , the DefaultURLResolverService becomes obsolete.
Therefore, removing the service and its implementation for clarity.

### When should this PR be merged

after https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1353

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilderTest.java [hunk 1]
```diff
@@ -35,7 +35,6 @@
 import org.wso2.carbon.identity.base.IdentityConstants;
 import org.wso2.carbon.identity.core.ServiceURLBuilder;
 import org.wso2.carbon.identity.core.URLBuilderException;
-import org.wso2.carbon.identity.core.URLResolverService;
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilderTest.java:38]: Do we need this test class now?

Comment (ChamathNS) [reply components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilderTest.java:38]: Yeah. This test class is for the new API.

**Comment by thanujalk** [comment]
@ChamathNS can we make this a PR and merge?

**Comment by ChamathNS** [comment]
> @ChamathNS can we make this a PR and merge?

we can merge this only after merging wso2-extensions/identity-inbound-auth-oauth#1353

**Pull request merged**